src/main/java/com/salessparrow/api/lib/GetCrmActionSuggestions.java

```diff
@@ -0,0 +1,116 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.salessparrow.api.controllers.SuggestionsController;
+import com.salessparrow.api.dto.entities.AddTaskSuggestionEntityDto;
+import com.salessparrow.api.dto.formatter.CrmActionSuggestionsFormatterDto;
+import com.salessparrow.api.exception.CustomException;
+import com.salessparrow.api.lib.errorLib.ErrorObject;
+import com.salessparrow.api.lib.openAi.OpenAiPayloadBuilder;
+import com.salessparrow.api.lib.openAi.OpenAiRequest;
+
+/**
+ * GetCrmActionSuggestions is a class for getting the crm action suggestions.
+ */
+@Component
+public class GetCrmActionSuggestions {
+  @Autowired
+  private OpenAiRequest openAiRequest;
+
+  @Autowired
+  private OpenAiPayloadBuilder openAiPayloadBuilder;
+
+  private Logger logger = org.slf4j.LoggerFactory.getLogger(SuggestionsController.class);
+  
+  /**
+   * Get the crm action suggestions.
+   * @param text
+   * @return
+   */
+  public CrmActionSuggestionsFormatterDto getTaskSuggestions(String text) {
+    logger.info("Crm actions suggestions lib called");
+    
+    String escapedText = escapeForJson(text);
+    String payload = openAiPayloadBuilder.payloadForCrmActionsSuggestions(escapedText);
+    
+    String response = openAiRequest.makeRequest(payload).getResponseBody();
+
+    return parseResponse(response);
+  }
+
+  /**
+   * Parse the response from openai.
+   * @param responseBody
+   * @return
+   */
+  private CrmActionSuggestionsFormatterDto parseResponse(String responseBody){
+    CrmActionSuggestionsFormatterDto crmActionSuggestionsFormatterDto = new CrmActionSuggestionsFormatterDto();
+    
+    try {
+      Util util = new Util();
+      JsonNode rootNode = util.getJsonNode(responseBody);
+      JsonNode argumentsNode = rootNode.get("choices").get(0).get("message").get("function_call").get("arguments");  
+
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+      String argumentsJson = objectMapper.convertValue(argumentsNode, String.class);
+
+      Map<String, List<AddTaskSuggestionEntityDto>> arguments = objectMapper.readValue(argumentsJson, new TypeReference<Map<String, List<AddTaskSuggestionEntityDto>>>() {});
+      List<AddTaskSuggestionEntityDto> addTaskList = arguments.get("add_task");
+
+
+      List<AddTaskSuggestionEntityDto> formattedTaskSuggestionEntityDtos = new ArrayList<>();
+      if (addTaskList != null) {
+        for (AddTaskSuggestionEntityDto addTask : addTaskList) {
+          AddTaskSuggestionEntityDto addTaskSuggestionEntityDto = new AddTaskSuggestionEntityDto();
+          addTaskSuggestionEntityDto.setDescription(addTask.getDescription());
+
+          // Format the response check if duedate format is YYYY-MM-DD else remove duedate
+          String dueDate = addTask.getDueDate();
+          if (dueDate != null && dueDate.length() > 0) {
+            Pattern pattern = Pattern.compile("\\d{4}-\\d{2}-\\d{2}");
```

Comment:
Use a date function for validation. The date validation for create task api can be used as the formats should be same.